crates/matrix-sdk-crypto/src/verification/sas/mod.rs

```diff
@@ -182,37 +181,15 @@ impl Sas {
     ///
     /// Returns the new `Sas` object and a `StartEventContent` that needs to be
     /// sent out through the server to the other device.
-    #[allow(clippy::too_many_arguments)]
     pub(crate) fn start(
-        private_identity: PrivateCrossSigningIdentity,
-        other_device: ReadOnlyDevice,
-        store: VerificationStore,
-        own_identity: Option<ReadOnlyOwnUserIdentity>,
-        other_identity: Option<ReadOnlyUserIdentities>,
+        identities: IdentitiesBeingVerified,
         transaction_id: Option<OwnedTransactionId>,
```

Comment:
One possibility would be to make this non-optional, so callers have to pass `TransactionId::new` instead of `None`, but also callers that pass `Some()` right now can omit that.

Reply:
Did so in https://github.com/matrix-org/matrix-rust-sdk/pull/632/commits/2a5f17005dd64db52300e716a5a6f15da83e644a.